### PR TITLE
kile-wl: unstable-2021-04-22 -> unstable-2021-06-01

### PR DESCRIPTION
**Pull request by devins2518**
###### Motivation for this change
Kile-wl has had a lot of progress upstream, needed version bump.

###### Things done
Updated package to latest master commit

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I have an nvidia card so I'm not able to test the binary, but it should work.

###### Maintainer pings

cc @fortuneteller2k

### CODE REVIEW

File: pkgs/applications/misc/kile-wl/default.nix [hunk 1]
```diff
@@ -2,19 +2,19 @@
...
+  version = "unstable-2021-06-01";
 
   src = fetchFromGitLab {
     owner = "snakedye";
     repo = "kile";
-    rev = "b97b9f1e5b33862b33918efaf23fd1c0c5d7058a";
-    sha256 = "sha256-97qJd3o8nJt8IX5tyGWtAmJsIv5Gcw1xoBFwxAqk7I8=";
+    rev = "5d8dcbdb908e1f8bdc37a731c6e1c5d8c2ace3ba";
+    sha256 = "sha256-f3S91Ui6SKcbkmlHqqjlx4M/Te+2no75W3eAVKd769s=";
   };
```
Comment (fortuneteller2k) [inline pkgs/applications/misc/kile-wl/default.nix:13]:
```suggestion

nativeBuildInputs = [ scdoc ];
```

Comment (fortuneteller2k) [reply pkgs/applications/misc/kile-wl/default.nix:13]: and add `scdoc` to the arguments

Comment (devins2518) [reply pkgs/applications/misc/kile-wl/default.nix:13]: I've added that but nothing else is showing up in result/. Do I need to add a post install hook or something?

Comment (fortuneteller2k) [reply pkgs/applications/misc/kile-wl/default.nix:13]: yeah sure

Comment (devins2518) [reply pkgs/applications/misc/kile-wl/default.nix:13]: Ok, got that figured out, it should be installed to result/usr/share/man/man1 right?

Comment (fortuneteller2k) [reply pkgs/applications/misc/kile-wl/default.nix:13]: `result/share/man/man1`, drop the usr

Comment (devins2518) [reply pkgs/applications/misc/kile-wl/default.nix:13]: Gotcha

Comment (devins2518) [reply pkgs/applications/misc/kile-wl/default.nix:13]: Fixed in 4d5788b. Also updated the Cargo.lock.

**Review by moni-dz — APPROVED** [review]

**Review by moni-dz — APPROVED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125196` at 4d5788be run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/63d2c1db0c0e6ed96747b5a8782f8495">kile-wl</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/applications/misc/kile-wl/default.nix [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
   cargoPatches = [ ./update-Cargo-lock.diff ];
 
-  cargoSha256 = "sha256-TEgIiw/XTDUOe9K7agHWI86f88w+eDJ332V0CgNHtfo=";
+  cargoSha256 = "sha256-FcmYQ9e+oUz1WGE8pI/F8gm/y/nFS0O7W5FiUWpY3Js=";
+
+  nativeBuildInputs = [ scdoc ];
+
+  postInstall = ''
+    mkdir -p "$out/usr/share/man/man1"
+    install -Dm644 doc/*.1.gz "$out/share/man/man1/"
```
Comment (fortuneteller2k) [inline pkgs/applications/misc/kile-wl/default.nix:23]:
```suggestion
    mkdir -p $out/usr/share/man
    scdoc < doc/kile.1.scd > $out/usr/share/man/kile.1
```

**Review by moni-dz — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/kile-wl/default.nix (hunk 2 above)
Comment (devins2518) [reply pkgs/applications/misc/kile-wl/default.nix]: Fixed in 6729411.

**Review by moni-dz — APPROVED** [review]

**Review by legendofmiracles — APPROVED** [review]

**Review by Artturin — APPROVED** [review]

File: pkgs/applications/misc/kile-wl/default.nix [hunk 3]
```diff
@@ -1,20 +1,27 @@
-{ lib, fetchFromGitLab, rustPlatform }:
+{ lib, fetchFromGitLab, rustPlatform, scdoc }:
 
 rustPlatform.buildRustPackage rec {
   pname = "kile-wl";
-  version = "unstable-2021-04-22";
+  version = "unstable-2021-06-01";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/kile-wl/default.nix:5]: Please add something like https://github.com/NixOS/nixpkgs/blob/f59fb4b41ecd01ed4868922056ddfad6de4ccd94/pkgs/applications/audio/littlegptracker/default.nix#L47 to enable auto updates.

Comment (devins2518) [reply pkgs/applications/misc/kile-wl/default.nix:5]: Added in b75441c.

**Review by moni-dz — APPROVED** [review]

File: pkgs/applications/misc/kile-wl/default.nix [hunk 4]
```diff
@@ -1,20 +1,31 @@
...
     owner = "snakedye";
     repo = "kile";
-    rev = "b97b9f1e5b33862b33918efaf23fd1c0c5d7058a";
-    sha256 = "sha256-97qJd3o8nJt8IX5tyGWtAmJsIv5Gcw1xoBFwxAqk7I8=";
+    rev = "5d8dcbdb908e1f8bdc37a731c6e1c5d8c2ace3ba";
+    sha256 = "sha256-f3S91Ui6SKcbkmlHqqjlx4M/Te+2no75W3eAVKd769s=";
+  };
+
+  passthru.updateScript = unstableGitUpdater {
+    url = "https://gitlab.com/snakedye/kile.git";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/kile-wl/default.nix:15]: That won't update the cargo.lock, right?

Comment (legendofmiracles) [reply pkgs/applications/misc/kile-wl/default.nix:15]: Cargo.lock is ignored by upstream

**Comment by devins2518** [comment]
Upstream added the Cargo.lock so I've updated the revision and removed the patch

File: pkgs/applications/misc/kile-wl/default.nix (hunk 4 above)
Comment (devins2518) [reply pkgs/applications/misc/kile-wl/default.nix:15]: This should be fixed now that upstream is tracking Cargo.lock.

Comment (fortuneteller2k) [reply pkgs/applications/misc/kile-wl/default.nix:15]: I'm not sure if this updates the `cargoSha256`, does it?

**Review by moni-dz — APPROVED** [review]

File: pkgs/applications/misc/kile-wl/default.nix (hunk 4 above)
Comment (devins2518) [reply pkgs/applications/misc/kile-wl/default.nix:15]: True, is there a way to bypass that or update it in the script?

Comment (SuperSandro2000) [reply pkgs/applications/misc/kile-wl/default.nix:15]: nix-update can do that.

**Comment by snakedye** [comment]
Update on this pr?

**Pull request merged**